addons/mrp_subcontracting/models/stock_move.py

```diff
@@ -151,6 +163,48 @@ def _action_record_components(self):
             'context': dict(self.env.context, subcontract_move_id=self.id),
         }
 
+    def action_serial_mass_receipt(self):
+        self.ensure_one()
+        production = self.move_orig_ids.production_id[-1:]
+        dummy, dummy, missing_components, dummy = production._check_serial_mass_produce_components()
+        if missing_components:
+            message = _("Make sure enough quantities of these components are reserved to carry on production:\n")
+            message += "\n".join(component.name for component in missing_components)
+            pickings = self.env['stock.picking'].search([
+                ('group_id', '=', production.procurement_group_id.id),
+                ('state', 'not in', ('done', 'cancel'))
+            ])
+            purchase_orders = self.env['purchase.order'].search([
+                ('group_id', '=', production.procurement_group_id.id),
+                ('state', 'not in', ('purchase', 'done', 'cancel'))
+            ])
```

Comment:
Subcontracting does not depend on the purchase module and a PO is optional for subcontracting (can be done directly with a receipt Receive From = subcontractor, Product = subcontracted product), so this will throw a server error when user clicks on button without purchase installed

Reply:
well seen, adapted